src/api/src/lib.rs

```diff
@@ -71,52 +66,14 @@ impl CLI {
             format!("Invalid command line arguments: {}", e)
         })?;
 
-        Ok(VMMConfig {
-            memory_config: MemoryConfig::try_from(
-                matches
-                    .value_of("memory")
-                    .expect("Missing memory configuration")
-                    .to_string(),
-            )
-            .map_err(|e| {
-                eprintln!("{}", help_msg);
-                format!("{}", e)
-            })?,
-            kernel_config: KernelConfig::try_from(
-                matches
-                    .value_of("kernel")
-                    .expect("Missing kernel configuration")
-                    .to_string(),
-            )
-            .map_err(|e| {
-                eprintln!("{}", help_msg);
-                format!("{}", e)
-            })?,
-            vcpu_config: VcpuConfig::try_from(
-                matches
-                    .value_of("vcpu")
-                    .expect("Missing vCPU configuration")
-                    .to_string(),
-            )
-            .map_err(|e| {
-                eprintln!("{}", help_msg);
-                format!("{}", e)
-            })?,
-            block_config: match matches.value_of("block") {
-                Some(value) => Some(BlockConfig::try_from(value.to_string()).map_err(|e| {
-                    eprintln!("{}", help_msg);
-                    format!("{}", e)
-                })?),
-                None => None,
-            },
-            network_config: match matches.value_of("net") {
-                Some(value) => Some(NetConfig::try_from(value.to_string()).map_err(|e| {
-                    eprintln!("{}", help_msg);
-                    format!("{}", e)
-                })?),
-                None => None,
-            },
-        })
+        Ok(VMMConfig::builder()
+            .memory_config(matches.value_of("memory"))
+            .kernel_config(matches.value_of("kernel"))
+            .vcpu_config(matches.value_of("vcpu"))
+            .network_config(matches.value_of("network"))
```

Comment:
"network" should be "net". The sad thing is there was no test to catch this thing :( (sorry for that). Is it possible to add a test for this or include it in some other one (for example [this test](https://github.com/rust-vmm/vmm-reference/blob/master/src/api/src/lib.rs#L257))?

Reply:
Oh. Good catch! Fixed.

Reply:
I have added a separate commit that consistently uses `net_config`. Hope this is fine, if this is not required, We can simply revert that commit.

Reply:
Looks good to me; the commit message is a bit misleading, I think it should've been: We were using the network config param as **`network_config`**...

Regarding the other changes, I'll have a look on Monday. Thanks!